### PR TITLE
Options: merge SCoreStartupParameter into SConfig

### PR DESCRIPTION
**Pull request by degasus**
There is no need to have a struct within a struct just to group our variables.

This PR merges both structs into one place.

Please review this one soon as it will have lots of merge conflicts on rebasing.

Based on PR #2592

### CODE REVIEW

**Comment by JosJuice** [comment]
👏

File: Source/Core/DolphinWX/Cheats/CheatsWindow.cpp [hunk 1]
```diff
@@ -154,7 +153,7 @@ void wxCheatsWindow::OnEvent_Close(wxCloseEvent& ev)
 void wxCheatsWindow::UpdateGUI()
 {
 	// load code
-	SCoreStartupParameter parameters = SConfig::GetInstance().m_LocalCoreStartupParameter;
+	SConfig& parameters = SConfig::GetInstance();
```
Comment (degasus) [inline Source/Core/DolphinWX/Cheats/CheatsWindow.cpp]: Please keep care about this line on reviewment.

Comment (JosJuice) [inline Source/Core/DolphinWX/Cheats/CheatsWindow.cpp]: `UpdateGUI` only calls const methods on `parameters`, so making this a reference doesn't lead to any difference in behavior.

File: Source/Core/Core/ConfigManager.h [hunk 2]
```diff
@@ -36,7 +47,140 @@ struct SConfig : NonCopyable
 	std::vector<std::string> m_ISOFolder;
 	bool m_RecursiveISOFolder;
 
-	SCoreStartupParameter m_LocalCoreStartupParameter;
+	// Settings
+	bool bEnableDebugging;
+	#ifdef USE_GDBSTUB
+	int  iGDBPort;
```
Comment (Sonicadvance1) [inline Source/Core/Core/ConfigManager.h]: Double space!

**Comment by Sonicadvance1** [comment]
Other than that comment, lgtm.

**Pull request merged**